test/integration/role_test.lua

```diff
@@ -0,0 +1,66 @@
+local fio = require('fio')
+local t = require('luatest')
+local helpers = require('test.helper')
+local g = t.group('expirationd_intergration_role')
+local is_cartridge_helpers, cartridge_helpers = pcall(require, 'cartridge.test-helpers')
+
+g.before_all(function(cg)
+    if is_cartridge_helpers then
+        local entrypoint_path = fio.pathjoin(helpers.project_root,
+                                             'test',
+                                             'entrypoint',
+                                             'srv_role.lua')
+        cg.cluster = cartridge_helpers.Cluster:new({
+            datadir = fio.tempdir(),
+            server_command = entrypoint_path,
+            use_vshard = true,
+            replicasets = {
+                {
+                    uuid = helpers.uuid('a'),
+                    alias = 'router',
+                    roles = { 'vshard-router' },
+                    servers = {
+                        { instance_uuid = helpers.uuid('a', 1), alias = 'router' },
+                    },
+                },
+                {
+                    uuid = helpers.uuid('b'),
+                    alias = 's-1',
+                    roles = { 'vshard-storage', 'customers-storage', 'expirationd' },
```

Comment:
expirationd may indeed fail to start on a non-existent space, but it should be fixed:
https://github.com/tarantool/expirationd/issues/116

So this may be a problem if the custom role creates a space that will used by expirationd in apply_config().

Reply:
Yep, this must be fixed berofe releasing next verion of expirationd.
On the moment of role start space may not be existed because it will be created later, for exapmle by firing migrations. Thus this must be fixed anyway. So, seems that roles logic must be designed based on #116 will be fixed, IMHO.

Reply:
It should be fixed in #126